crates/bevy_core_pipeline/src/mipmap_generator.rs

```diff
@@ -0,0 +1,620 @@
+//! A wrapper that runs a shader repeatedly to populate a series of mipmaps.
+//!
+//! A variety of passes need to or will need to do this, and using this module
+//! helps to reduce boilerplate and code duplication.
+
+use std::{borrow::Cow, hash::Hash, marker::PhantomData, sync::Arc};
+
+use bevy_app::{App, Plugin};
+use bevy_asset::Handle;
+use bevy_ecs::{
+    component::Component,
+    system::Resource,
+    world::{FromWorld, World},
+};
+use bevy_math::UVec2;
+use bevy_render::{
+    render_resource::{
+        AddressMode, AsBindGroup, BindGroup, BindGroupEntry, BindGroupLayout,
+        BindGroupLayoutDescriptor, BindGroupLayoutEntry, BindingResource, BindingType,
+        CachedRenderPipelineId, ColorTargetState, ColorWrites, Extent3d, FilterMode, FragmentState,
+        LoadOp, MultisampleState, Operations, PipelineCache, PreparedBindGroup, PrimitiveState,
+        RenderPassColorAttachment, RenderPassDescriptor, RenderPipelineDescriptor, Sampler,
+        SamplerBindingType, SamplerDescriptor, Shader, ShaderDefVal, ShaderStages,
+        SpecializedRenderPipeline, SpecializedRenderPipelines, TextureDescriptor, TextureDimension,
+        TextureFormat, TextureSampleType, TextureUsages, TextureView, TextureViewDescriptor,
+        TextureViewDimension,
+    },
+    renderer::{RenderContext, RenderDevice},
+    texture::{CachedTexture, TextureCache},
+    RenderApp,
+};
+
+use crate::fullscreen_vertex_shader::fullscreen_shader_vertex_state;
+
+/// All mipmappers implement this trait.
```

Comment:
Ditto here: a simple explanation of what a mipmap actually is would be helpful to rendering beginners.